### PR TITLE
Update KernelExceptionSubscriber in case several sites share the same domain

### PR DESCRIPTION
**Pull request by ngilain**
https://github.com/open-orchestra/open-orchestra-front-bundle/pull/122
https://github.com/open-orchestra/open-orchestra-model-bundle/pull/415

### CODE REVIEW

**Comment by nicolasThal** [comment]
A general remark, I would not base my private method on some base parameters that have been created in a previous private method from the class.

I think that will grow the mess, and make all properties really difficult to track

File: FrontBundle/EventSubscriber/KernelExceptionSubscriber.php [hunk 1]
```diff
@@ -50,57 +53,77 @@ public function __construct(
...
+
+        if ($node) {
+            return $this->templating->render(
+                'OpenOrchestraFrontBundle:Node:show.html.twig',
+                array(
+                    'node' => $node,
+                    'parameters' => array('siteId' => $node->getSiteId(), '_locale' => $node->getLanguage())
+                )
+            );
+        } else {
```
Comment (nicolasThal) [inline FrontBundle/EventSubscriber/KernelExceptionSubscriber.php]: the else is useless

File: FrontBundle/EventSubscriber/KernelExceptionSubscriber.php [hunk 2]
```diff
@@ -50,57 +53,77 @@ public function __construct(
...
-                    $html = $this->templating->render(
-                        'OpenOrchestraFrontBundle:Node:show.html.twig',
-                        array(
-                            'node' => $node,
-                            'parameters' => array('siteId' => $node->getSiteId(), '_locale' => $node->getLanguage())
-                        )
-                    );
-                    $event->setResponse(new Response($html, 404));
-                }
+            if ($this->currentSite && $this->currentLanguage && $html = $this->getCustom404Html()) {
```
Comment (nicolasThal) [inline FrontBundle/EventSubscriber/KernelExceptionSubscriber.php]: the only real check is if $html is null or not

File: FrontBundle/EventSubscriber/KernelExceptionSubscriber.php [hunk 3]
```diff
@@ -50,57 +53,77 @@ public function __construct(
...
+        foreach ($matchingSites as $site) {
             foreach ($site->getAliases() as $alias) {
-                if ($host == $alias->getDomain() && strpos($path . '/', $alias->getPrefix() . '/') === 0) {
-                    $targetAlias = $alias;
-                    break;
+                if ($alias->getPrefix() == '') {
+                    $possibleSite = $site;
+                    $possibleAlias = $alias;
+                } else {
+                    if ($host == $alias->getDomain() && strpos($path . '/', trim($alias->getPrefix(), '/') . '/') === 0) {
```
Comment (nicolasThal) [inline FrontBundle/EventSubscriber/KernelExceptionSubscriber.php]: I think that your check is not deep enough.

We wanted to find the longest prefix available.

For instance with the url  /en/domain/edit : 
- site 1 : prefix en
- site 2 : prefix en/domain

We should return the site 2

**Comment by ngilain** [comment]
@nicolasThal , @itkg-alavieille , @itkg-rboessel

File: FrontBundle/EventSubscriber/KernelExceptionSubscriber.php [hunk 4]
```diff
@@ -50,57 +50,103 @@ public function __construct(
...
+        if (!$site || !$language) {
+            return null;
+        }
+
+        $siteId = $site->getSiteId();
+        $nodeId = ReadNodeInterface::ERROR_404_NODE_ID;
+        $node = $this->nodeRepository->findOnePublishedByNodeIdAndLanguageAndSiteIdInLastVersion($nodeId, $language, $siteId);
+
+        if ($node) {
+            return $this->templating->render(
```
Comment (alavieille) [inline FrontBundle/EventSubscriber/KernelExceptionSubscriber.php:141]: add an empty line

**Comment by alavieille** [comment]
:+1:  with fix style

**Pull request merged**